### PR TITLE
vsl-69: Refactor send modal

### PR DESCRIPTION
**Pull request by linxiao-li**
This pr is part 1 of the initiative of vsl-69: fix UI discrepancy of send modal, refactor the send modal to allow adding more features required in the [Figma design](https://www.figma.com/file/E611fSwUIzNy3peFwxIfBa/PROJECT-VSL?node-id=529%3A409)

**What this pr includes**
1. All existing flow and functionalities of the old version
2. an additional cointype dropdown
3. UI fix for nft selector (https://linear.app/dappercollectives/issue/VSL-97/send-nft-dropdown-layout)
4. UI fix for clearing the modal content after clicking on the cancel button (https://linear.app/dappercollectives/issue/VSL-103/send-cancel-btn-should-clear-state)
5. allow for absolute path import
6. modified Dropdown reusable component to render with itemValue and displayText 

**What this pr doesn't include**
1. Max button on the amount input(coming soon)
2. Convert amount to usd (de-scoped)
3. Pulling the coin types and their quantities dynamically(coming soon)
4. Calculating network fee(coming soon)
7. calculating total value in USD (de-scoped)

sending tokens flow
Screen Shot 2022-07-27 at 4 01 37 PM
Screen Shot 2022-07-27 at 4 13 44 PM

sending nft flow
Screen Shot 2022-07-27 at 4 01 50 PM
Screen Shot 2022-07-27 at 4 13 07 PM

### CODE REVIEW

File: packages/client/src/components/SendTokens/components/AddressInput.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { useAddressValidation } from "hooks";
+import { Check } from "components/Svg";
+import { isAddr, formatAddress } from "utils";
+const AddressInput = () => {
+  const [sendModalState, setSendModalState] = useContext(SendTokensContext);
+  const web3 = useContext(Web3Context);
+  const { recipient, recipientValid } = sendModalState;
+  const { isAddressValid } = useAddressValidation(web3.injectedProvider);
+
+  const onrecipientChange = async (e) => {
```
Comment (KsenijaZivkovic) [inline packages/client/src/components/SendTokens/components/AddressInput.js:13]: nit: onRecipientChange

File: packages/client/src/components/SendTokens/components/AddressInput.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    let newValue = e.target.value;
+    let isValid = isAddr(e.target.value);
+    if (isValid) {
+      isValid = await isAddressValid(formatAddress(e.target.value));
```
Comment (KsenijaZivkovic) [inline packages/client/src/components/SendTokens/components/AddressInput.js:17]: Can these validations for address be combined?

File: packages/client/src/components/SendTokens/components/AmountInput.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          value={tokenAmount}
+          onChange={(e) =>
+            setSendModalState((prevState) => ({
+              ...prevState,
+              tokenAmount: e.target.value,
+            }))
+          }
+        />
+
+        {/* <div style={{ position: "absolute", top: "20px", right: "0px" }}>
```
Comment (KsenijaZivkovic) [inline packages/client/src/components/SendTokens/components/AmountInput.js:24]: Can we delete the comment?

File: packages/client/src/constants/maps.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { ASSET_TYPES, COIN_TYPES, SAFE_TYPES } from "./enums";
+export const ASSET_TYPE_TO_META = {
+  [ASSET_TYPES.TOKEN]: {
+    displayName: "Coins",
```
Comment (KsenijaZivkovic) [inline packages/client/src/constants/maps.js:4]: Will here in the future be more properties besides display name?

**Review by amyliu6 — APPROVED** [review]
LGTM! Great job on refactoring! 👍

File: packages/client/src/components/SendTokens/components/AmountInput.js (hunk 3 above)
Comment (linxiao-li) [reply packages/client/src/components/SendTokens/components/AmountInput.js:24]: Yes, the functions to support these two button will be in the next pr

File: packages/client/src/constants/maps.js (hunk 4 above)
Comment (linxiao-li) [reply packages/client/src/constants/maps.js:4]: Exactly! It's intend to hold all config data of different types of coins, for example, the url for the thumbnail of the coin

**Review by KsenijaZivkovic — APPROVED** [review]

**Pull request merged**